tests/test_url_shortening.py

```diff
@@ -0,0 +1,206 @@
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+
+    def test_anonymous_user_shorten_via_api(self):
+        """Test that anonymous users cannot shorten urls via the API."""
+        response = self.client.post(url_for('api.shorten_url'),
+                                    data=self.post_data)
+        json_response = json.loads(response.data.decode('utf-8'))
+        msg = json_response['message']
+        self.assertEqual(msg, 'Permission required! You are not allowed'
+                         ' access to this resource.')
+        self.assertEqual(403, response.status_code)
+
+    def test_registered_user_shorten_via_api(self):
+        """Test that registered users can shorten urls via the API."""
+        self.user.save()
+        email_header = self.api_email_auth_headers('ichiato@yahoo.com',
+                                                   'password')
+        token_response = self.client.get(url_for('api.get_token'),
+                                         headers=email_header)
+        json_token_response = json.loads(token_response.data.decode('utf-8'))
+        token = json_token_response['token']
+        token_header = self.api_token_auth_headers(token)
+        shorten_response_with_email_auth = self.client.post(
+            url_for('api.shorten_url'), headers=email_header,
+            data=self.post_data)
+        shorten_response_with_token_auth = self.client.post(
+                            url_for('api.shorten_url'), headers=token_header,
+                            data=self.post_data)
+        json_shorten_response_with_email = json.loads(
+            shorten_response_with_email_auth.data.decode('utf-8'))
+        email_msg = json_shorten_response_with_email['message']
+        json_shorten_response_with_token = json.loads(
+            shorten_response_with_token_auth.data.decode('utf-8'))
+        self.assertEqual(email_msg, 'Permission required! You are not allowed'
+                         ' access to this resource.')
+        self.assertIn('short_url', json_shorten_response_with_token)
+        self.assertEqual(403, shorten_response_with_email_auth.status_code)
+        self.assertEqual(201, shorten_response_with_token_auth.status_code)
+
+    def test_registered_user_with_no_body(self):
+        """Test that anonymous users cannot shorten urls via the API."""
+        self.user.save()
+        email_header = self.api_email_auth_headers('ichiato@yahoo.com',
+                                                   'password')
+        token_response = self.client.get(url_for('api.get_token'),
+                                         headers=email_header)
+        json_token_response = json.loads(token_response.data.decode('utf-8'))
+        token = json_token_response['token']
+        token_header = self.api_token_auth_headers(token)
+        shorten_response_with_token_auth = self.client.post(
```

Comment:
Again aim for one response per method, and you would have no need for these long names you are currently using.